src/parser/content-evaluator-module.ts

```
@@ -594,12 +610,85 @@ export class ContentEvaluatorModule extends BaseModule {
+  }
+
+  /**
+   * Legacy single-prompt generator for issue comments.
+   * Used internally by chunk estimation logic to determine the max token size.
+   */
   _generatePromptForComments(issue: string, username: string, allComments: AllComments) {
+    const relevance = this._generateSpecializedIssuePrompt(issue, username, allComments, "relevance");
+    const helpfulness = this._generateSpecializedIssuePrompt(issue, username, allComments, "helpfulness");
+    const research = this._generateSpecializedIssuePrompt(issue, username, allComments, "research");
+    const prompts = [relevance, helpfulness, research];
+    return prompts.reduce((a, b) => (a.length > b.length ? a : b));
```

ESLint

[error] Add an initial value to this "reduce()" call.

(sonarjs/reduce-initial-value)